bindings/rust/s2n-tls/src/connection.rs

```diff
@@ -972,6 +972,24 @@ impl Connection {
         }
     }
 
+    pub fn kem_name(&self) -> Result<Option<&str>, Error> {
```

Comment:
How should we decide if an API returns a `Result<Option>` or just an `Option`? It seems like we have APIs that do both. Currently this API won't ever fail, and a failure might be communicated pretty well with None. So is this a case where just an `Option` might be preferable over a `Result<Option>`?